Vagrantfile

```diff
@@ -252,27 +234,36 @@ Vagrant.configure(2) do |config|
   end
 
   # AWS Provider.  Set enviornment variables for values below to use
-  config.vm.provider "aws" do |aws, override|
-    override.vm.box               = "dummy"
-    override.nfs.functional       = false
-    override.ssh.username         = "vagrant"
-    override.ssh.private_key_path = ENV['AWS_PRIVATE_KEY_PATH']
-    aws.instance_type             = "m3.xlarge"
-    aws.block_device_mapping      = [{ 'DeviceName' => '/dev/sda1', 'Ebs.VolumeSize' => 32 }]
-    aws.access_key_id             = ENV['AWS_ACCESS_KEY_ID']
-    aws.secret_access_key         = ENV['AWS_SECRET_ACCESS_KEY']
-    aws.keypair_name              = ENV['AWS_KEYPAIR_NAME']
-    aws.security_groups           = ENV['AWS_SECURITY_GROUP']
-    aws.ami                       = ENV['AWS_AMI_UBUNTU1404']
+  config.vm.provider :aws do |aws, override|
+    override.nfs.functional  = false
+    aws.instance_type        = ENV.fetch('AWS_INSTANCE_TYPE', 'm3.xlarge')
+    aws.block_device_mapping = [{ 'DeviceName' => '/dev/sda1', 'Ebs.VolumeSize' => 32 }]
+
+    # By default we assume that we're using the AWS "dummy" box and manually
+    # specify an Ubuntu AMI.  These assumptions will be removed when we migrate
+    # to boxes that have native AWS providers that use the standard Vagrant key.
+    if ENV.fetch('AWS_DUMMY_BOX', 'yes') == 'yes'
+      override.vm.box = 'dummmy'
+      override.ssh.private_key_path = ENV['AWS_PRIVATE_KEY_PATH']
+      aws.ami = ENV['AWS_AMI_UBUNTU1404']
+    end
+
+    if ENV.key?('AWS_KEYPAIR_NAME')
+      aws.keypair_name = ENV['AWS_KEYPAIR_NAME']
+    end
+
+    if ENV.key?('AWS_SECURITY_GROUP')
+      aws.security_groups = ENV['AWS_SECURITY_GROUP']
+    end
+
+    aws.tags = {
+      'Name' => ENV.fetch('AWS_INSTANCE_NAME_TAG', 'jenkins-hootenanny'),
+      'URL'  => ENV.fetch('AWS_INSTANCE_URL_TAG', 'https://github.com/ngageoint/hootenanny'),
+    }
 
     # Copy over predownloaded packages if they are found
     override.vm.provision "software", type: "shell", run: "always", :inline => "( [ -d /home/vagrant/hoot/software ] && cp /home/vagrant/hoot/software/* /home/vagrant ) || true"
-    override.vm.provision "hoot", type: "shell", :privileged => false, :path => "VagrantProvision.sh"
-    override.vm.provision "build", type: "shell", :privileged => false, :path => "VagrantBuild.sh"
     override.vm.provision "EGD", type: "shell", :privileged => false, :inline  => "([ -f ~/ActivateEGDplugin.sh ] && sudo -u tomcat8 ~/ActivateEGDplugin.sh /var/lib/tomcat8) || true"
-    override.vm.provision "tomcat", type: "shell", :privileged => false, :inline => "sudo service tomcat8 restart", run: "always"
-    override.vm.provision "mapnik", type: "shell", :privileged => false, :inline => "sudo service node-mapnik-server start", run: "always"
-    override.vm.provision "hadoop", type: "shell", :privileged => false, :inline => "stop-all.sh && start-all.sh", run: "always"
```

Comment:
I believe I overrode these existing provisioners because the order mattered and when I only added the provision steps unique to the AWS vagrant machine, the provision steps executed in a different order.  I'll check the output of the currently running job to see what order they happened in.

Reply:
The problem with that approach is it makes it _impossible_ to customize the provisioner based on OS.  In other words, you can only use the override hack to change script once, if we set the `VagrantProvisionCentOS7.sh` here then it'll be used for Ubuntu as well on AWS.

I think we can prevent this by adding noop `software` and `EGD` provisioners to the individual box configurations.

Reply:
Confirmed that even overridden provision scripts are added to the end of the provision order instead of kept in order.   All provision scripts need to be set up in the override, and the override must happen at the level of the individual box, or all of those settings will be obliterated by the global override at the end of the `Vagrantfile`.  This is going to take some work and time to figure out, but _maybe_ using a defined function called from the box definition level will workl to set up the proper overrides.

Reply:
Perhaps we pull the provisioner overrides out of the definition of the aws provider block and introduce two new definitions of vagrant machines, something like "aws_ubuntu_1404" and "aws_centos_7".  Then each machine can have it's own set of overrides and call the appropriate provisioner.